pipeline/builder.go

```diff
@@ -37,11 +38,18 @@ func build(c pipelineBuilderOptions) (Pipeline, error) {
 	}
 
 	switch {
-	case c.batchDecoder != nil:
-		setupBatchProcessor(c, p)
-	case c.decoder != nil:
-		setupProcessor(c, p)
+	case c.messagePool != nil:
+		err = setupMessagePool(c, p)
+	case len(c.inputStreams) > 0:
+		err = setupStreamProcessor(c, p)
+	default:
+		err = errors.E(op, "invalid pipeline configuration")
```

Comment:
Nit: usually, I see the word "bad" instead of "invalid" to make error messages shorter